libs/types/src/permissions.rs

```diff
@@ -110,6 +111,12 @@ pub struct PermissionedCurrencyHolderInfo {
 pub struct TrancheInvestorInfo<TrancheId> {
 	tranche_id: TrancheId,
 	permissioned_till: Seconds,
+	is_frozen: bool,
```

Comment:
NOTE: This will require a migration because it affect the `Storage` type in `pallet_permissions`

Reply:
Luckily it does not in this case because the default value for `bool` is false which is exactly what we want to have for each tranche investor.

Reply:
But it will no try to deserialize more bytes that the stored bytes?

Reply:
I mean, the encode len has changed for this type

Reply:
> But it will no try to deserialize more bytes that the stored bytes?

I was quite sure that the deserialization would fill the missing bytes with the default values as long as the stored bytes can be decoded. However, the `try_decode_entire_state` check of `try-runtime CLI` proved me wrong and you right:

```
[2024-08-08T16:03:32Z ERROR runtime::executive] `try_decode_entire_state` failed with 20 errors
[2024-08-08T16:03:32Z ERROR runtime::executive] - 0. error: `Permissions::Permission` key `0xc2b6ac49ee131be4de5527a2ccab4a67d8a233a5123f5da90f0e75b26ae0949120eb146f1c3931800a363a6551685f87bbbbbbbbbb9cc5e90e3b3af64bdaf62c37eeffcb000000000000210545564d00847ce2c37eea396ebc1e48657b2fe6d1004f67bdf600000000` is undecodable
...
```

Therefore, I will write yet another migration. Luckily, we only have 20 decoding issues and not hundreds :D

Reply:
Thanks for checking this William! Sorry for having to write another migration 😢, to many for a single week 😆

Reply:
BTW, luckily that now `try-runtime` supports that utility 🙌🏻

Reply:
Yes, absolutely! Migration done: [9de643f](https://github.com/centrifuge/centrifuge-chain/pull/1947/commits/9de643fc235ab41ffa53d86a3cb861547f0e1e1b)